packages/dds/tree/src/feature-libraries/chunked-forest/codec/compressedEncode.ts

```diff
@@ -436,56 +436,63 @@ export function encodeValue(
 	}
 }
 
-export class EncoderCache implements TreeShaper, FieldShaper {
-	private readonly shapesFromSchema: Map<TreeNodeSchemaIdentifier, NodeEncoder> = new Map();
+export class EncoderCache implements NodeEncodeBuilder, FieldEncodeBuilder {
```

Comment:
If we document what this thing is, I can be less picky about its names since they don't have to convey everything.
```suggestion
/**
 * Cache used to avoid duplicating equivalent shapes during a batch of encode operations.
 * Additionally provides any common contextual information.
 */
export class EncoderCache implements NodeEncodeBuilder, FieldEncodeBuilder {
```